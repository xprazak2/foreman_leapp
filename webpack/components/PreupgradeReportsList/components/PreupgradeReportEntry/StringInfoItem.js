import React from 'react';
import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';

import InfoItem from './InfoItem';
import { itemIteratorId } from './helpers';

const StringInfoItem = ({
  entry,
  attr,
  tooltipText,
  translate,
  mapAttr,
  elipsed,
}) => {
  const inner = (
    <span>{translate ? __(mapAttr(entry, attr)) : mapAttr(entry, attr)}</span>
  );
  const innerContent = elipsed ? (
    <EllipsisWithTooltip placement="top">{inner}</EllipsisWithTooltip>
  ) : (
    inner
  );

  return (
    <InfoItem itemId={itemIteratorId(entry, attr)} tooltipText={tooltipText}>
      {innerContent}
    </InfoItem>
  );
};

StringInfoItem.propTypes = {
  entry: PropTypes.object.isRequired,
  attr: PropTypes.string.isRequired,
  tooltipText: PropTypes.string,
  translate: PropTypes.bool,
  mapAttr: PropTypes.func,
  elipsed: PropTypes.bool,
};

StringInfoItem.defaultProps = {
  translate: false,
  mapAttr: (entry, attr) => entry[attr],
  elipsed: false,
  tooltipText: undefined,
};

export default StringInfoItem;
