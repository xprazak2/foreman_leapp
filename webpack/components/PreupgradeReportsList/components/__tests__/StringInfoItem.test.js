import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import StringInfoItem from '../StringInfoItem';

const fixtures = {
  'should render': {
    entry: { title: 'no title', severity: 'low', id: 5, preupgradeReportId: 7 },
    attr: 'title',
  },
};

describe('StringInfoItem', () =>
  testComponentSnapshotsWithFixtures(StringInfoItem, fixtures));
