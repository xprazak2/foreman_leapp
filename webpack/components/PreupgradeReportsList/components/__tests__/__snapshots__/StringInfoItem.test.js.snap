// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringInfoItem should render 1`] = `
<InfoItem
  itemId="5-7-title"
  tooltipText=""
>
  <span>
    no title
  </span>
</InfoItem>
`;
